source/common/http/conn_manager_impl.cc

```diff
@@ -470,6 +487,47 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
       connection_manager_.config_, *snapped_route_config_, connection_manager_.random_generator_,
       connection_manager_.runtime_, connection_manager_.local_info_);
 
+  if (!cached_route_.valid()) {
+    cached_route_.value(snapped_route_config_->route(*request_headers_, stream_id_));
+  }
+
+  // Check for WebSocket upgrade request if the route exists, and supports WebSockets.
+  if ((protocol == Protocol::Http11) && cached_route_.value()) {
```

Comment:
For someone that wants to just proxy web sockets, there is actually no reason that they need to install the router filter (or any filters at all). However, IIRC, if someone does not install the router filter, and there is no websocket match here, there will be no response at all. IMO, if there are no filters and we attempt to start filter iteration, the connection manager should return 404. Can you add a TODO to that effect?